### PR TITLE
fix:error handled in all the cases

### PR DESCRIPTION
**Pull request by Laeeqdev**

### CODE REVIEW

File: src/notification/service/notificationService.ts [hunk 1]
```diff
@@ -8,13 +8,18 @@ import { WebhookService } from "../../destination/destinationHandlers/webhookHan
 import { SESService } from "../../destination/destinationHandlers/sesHandler";
 import { SMTPService } from "../../destination/destinationHandlers/smtpHandler";
 import { EVENT_TYPE } from "../../common/types";
-
+import {error} from "winston";
```
Comment (vivek-devtron) [inline src/notification/service/notificationService.ts:11]: Please remove this unused import

File: src/notification/service/notificationService.ts [hunk 2]
```diff
@@ -31,118 +36,127 @@ class NotificationService {
...
-                    configsMap.set(id, false)
-                });
-            });
-
-            settingsResults.forEach((setting) => {
 
-                const configArray =  setting.config as any;
-                if (Array.isArray(configArray)) {
-                  const webhookConfig = configArray.filter((config) => config.dest === 'webhook');
+             let settingsResults=await this.notificationSettingsRepository.findByEventSource(event.pipelineType, event.pipelineId, event.eventTypeId, event.appId, event.envId, event.teamId)
```
Comment (vivek-devtron) [inline src/notification/service/notificationService.ts:91]: Please make this const

File: src/notification/service/notificationService.ts [hunk 3]
```diff
@@ -31,118 +36,127 @@ class NotificationService {
...
 
-                  if (webhookConfig.length) {
-                    const webhookConfigRepository = new WebhookConfigRepository();
-                    webhookConfig.forEach(config => {
-                        webhookConfigRepository.getAllWebhookConfigs().then((templateResults: WebhookConfig[]) => {
-                            const newTemplateResult = templateResults.filter((t) => t.id === config.configId);
+                 this.logger.info('notificationSettingsRepository.findByEventSource')
+               if (!settingsResults || settingsResults.length == 0) {
+                    this.logger.info("no notification settings found for event " + event.correlationId);
+                   return await new CustomError("no notification settings found for event",404)
```
Comment (vivek-devtron) [inline src/notification/service/notificationService.ts:96]: we can omit await

File: src/notification/service/notificationService.ts [hunk 4]
```diff
@@ -31,118 +36,127 @@ class NotificationService {
...
+
+                        if (webhookConfig.length) {
+                            const webhookConfigRepository = new WebhookConfigRepository();
+                            for (const config of webhookConfig) {
+                               let templateResults=await webhookConfigRepository.getAllWebhookConfigs()
+                                    const newTemplateResult = templateResults.filter((t) => t.id === config.configId);
+
+                                    if (newTemplateResult.length === 0) {
+                                        this.logger.info("no templates found for event ", event);
+                                        return await new CustomError("no templates found for event", 404);
```
Comment (vivek-devtron) [inline src/notification/service/notificationService.ts:125]: Same here

File: src/notification/service/notificationService.ts [hunk 5]
```diff
@@ -31,118 +36,127 @@ class NotificationService {
...
-                            h.handle(event, templateResults, setting, configsMap, destinationMap)
                         }
-                    })
+                    }
                 }
-            }
-            });
-        }).catch(err => this.logger.error("err" + err))
+           this.logger.info("notification sent");
+           return await new CustomError("notification sent",200)
```
Comment (vivek-devtron) [inline src/notification/service/notificationService.ts:156]: Wehy success is getting wrapped in custom error?

File: src/server.ts [hunk 6]
```diff
@@ -29,6 +29,7 @@ import { WebhookService } from './destination/destinationHandlers/webhookHandler
 import { WebhookConfig } from './entities/webhookconfig';
 import * as process from "process";
 import bodyParser from 'body-parser';
+import {CustomError} from "./entities/events";
```
Comment (vivek-devtron) [inline src/server.ts:32]: Please remove unused import

File: src/server.ts [hunk 7]
```diff
@@ -103,10 +104,10 @@ app.get('/test', (req, res) => {
     res.send('Test!');
 })
 
-app.post('/notify', (req, res) => {
+app.post('/notify', async(req, res) => {
     logger.info("notifications Received")
-    notificationService.sendNotification(req.body)
-    res.send('notifications sent')
+       let result=await notificationService.sendNotification(req.body);
```
Comment (vivek-devtron) [inline src/server.ts:109]: Please make it const

File: src/destination/destinationHandlers/smtpHandler.ts [hunk 8]
```diff
@@ -186,6 +187,7 @@ export class SMTPService implements Handler {
         } catch (error) {
             this.logger.error('SMTP sendNotification error', error)
             throw new Error('Unable to send SMTP notification');
```
Comment (vivek-devtron) [inline src/destination/destinationHandlers/smtpHandler.ts:189]: We need to remove this line otherwise next line won't get executed

**Review by vivek-devtron — APPROVED** [review]

**Comment by Laeeqdev** [comment]
created new PR

**Pull request closed without merging**